### PR TITLE
Fix mcp script to avoid follow

### PR DESCRIPTION
**Pull request by mvpatel2000**
# What does this PR do?

`follow_run_logs` occasionally disconnects. Instead, we need to wait for the run to complete and then get logs. Additionally, we want to add a timeout to kill daily tests if they don't finish in time.

Along for ride, adds `datasets` to nlp install which breaks install rn

# What issue(s) does this change relate to?

[CO-1722](https://mosaicml.atlassian.net/browse/CO-1722)

[CO-1722]: https://mosaicml.atlassian.net/browse/CO-1722?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -391,23 +391,6 @@ stage('Build') {
                 }
             }]
         }
-        else if (!isComposerImage) {
```
Comment (bandish-shah) [inline .ci/Jenkinsfile:394]: Would it make sense to do this in it's own PR?  Just thinking if anything goes wrong it would be easier to revert a specific commit.

**Comment by nik-mosaic** [comment]
Why does adding datasets break install?

**Comment by mvpatel2000** [comment]
> Why does adding datasets break install?

Not having datasets breaks install bc it is used in the lm eval code but not registered as an nlp dependency

**Review by nik-mosaic — APPROVED** [review]

**Review by bandish-shah — APPROVED** [review]
LGTM
